src/idpi/operators/time_operators.py

```diff
@@ -34,23 +52,149 @@ def delta(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
-        raise ValueError(msg)
-
-    result = field - field.shift(time=index.item())
+    nsteps = _nsteps(field.valid_time, dtime)
+    result = field - field.shift(time=nsteps)
     result.attrs = field.attrs
     return result
+
+
+def min(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute minimum aggregate for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the minimum.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
+
+    Returns
+    -------
+    xr.DataArray
+        The field minimum for the given time delta.
+
+    """
+    nsteps = _nsteps(field.valid_time, dtime)
+    return field.rolling(time=nsteps).min()
+
+
+def max(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute maximum aggregate for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the maximum.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
+
+    Returns
+    -------
+    xr.DataArray
+        The field maximum for the given time delta.
+
+    """
+    nsteps = _nsteps(field.valid_time, dtime)
+    return field.rolling(time=nsteps).max()
+
+
+def avg(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute average aggregate for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the average.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
+
+    Returns
+    -------
+    xr.DataArray
+        The field average for the given time delta.
+
+    """
+    nsteps = _nsteps(field.valid_time, dtime)
+    return field.rolling(time=nsteps).mean()
+
+
+def sum(field: xr.DataArray, dtime: np.timedelta64) -> xr.DataArray:
+    """Compute sum aggregate for a given delta in time.
+
+    Parameters
+    ----------
+    field : xr.DataArray
+        Field that contains the input data.
+    dtime : np.timedelta64
+        Time delta for which to evaluate the sum.
+
+    Raises
+    ------
+    ValueError
+        if dtime is not multiple of the field time step
+        or if the time step is not regular.
+
+    Returns
+    -------
+    xr.DataArray
+        The field sum for the given time delta.
+
+    """
+    nsteps = _nsteps(field.valid_time, dtime)
+    return field.rolling(time=nsteps).sum()
+
+
+def resample(field: xr.DataArray, period: np.timedelta64) -> xr.DataArray:
```

Comment:
I suggest dt instead of period. More generically (not only for time) would be sample interval

Reply:
I think interval is more explicit than dt